examples/auth/db/index.ts

```diff
@@ -1,15 +1,7 @@
 import {PrismaClient} from "@prisma/client"
 export * from "@prisma/client"
 
-let prisma: PrismaClient
-
-if (process.env.NODE_ENV === "production") {
-  prisma = new PrismaClient()
-} else {
-  // Ensure the prisma instance is re-used during hot-reloading
-  // Otherwise, a new client will be created on every reload
-  globalThis["prisma"] = globalThis["prisma"] || new PrismaClient()
-  prisma = globalThis["prisma"]
+export default function db(): PrismaClient {
```

Comment:
In order to allow for using the existing method to query the db, does it make sense to export the result of calling db?

Reply:
That's what we are doing currently and is a side-effect :D I.e. what we are trying to fix.

**But**, I just had an idea of using `Proxy` to make this side effect free but still use `db.model.etc`. Will experiment and report back.

Reply:
Thanks for clarifying, didn't know that was a side-effect.